xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -126,6 +150,272 @@ define(["jquery"], function ($) {
+   *   Undefined means current. Negative value removes property sort.
+   * @param {String} descending Specify whether the sort should be descending or not.
+   *   Undefined means toggle current direction
+   */
+  Logic.prototype.sort = function (property, level, descending) {
+    if (this.data.query.properties.indexOf(property) === -1) { return; }
+    // find property current sort level
+    var currentLevel = this.data.query.sort.findIndex(function (sortObject) {
+      return sortObject.property === property;
+    });
+    // default level
+    if (level === undefined) {
+      level = (currentLevel !== -1) ? currentLevel : 0;
+    } else if (level < 0) {
+      level = -1;
+    }
+    // default descending
+    if (descending === undefined) {
+      descending = (currentLevel !== -1) ? !this.data.query.sort[currentLevel].descending : false;
+    }
+    // create sort object
+    var sortObject = {
+      property: property,
+      descending: descending,
+    };
+    // apply sort
+    if (level !== -1) {
+      this.data.query.sort.splice(level, 1, sortObject);
+    }
+    if (currentLevel !== -1 && currentLevel !== level) {
+      this.data.query.sort.splice(currentLevel, 1);
+    }
+    // dispatch events
+    this._triggerEvent("sort", {
+      property: property,
+      level: level,
+      descending: descending,
+    });
+
+    // CALL FUNCTION TO FETCH NEW DATA HERE
+  };
+
+
+  /**
+   * Add new sort entry, shorthand of Logic.prototype.sort
+   * If the property is already sorting, does nothing
+   * @param {String} property The property to add to the sort
+   * @param {String} descending Specify whether the sort should be descending or not.
+   *   Undefined means toggle current direction
+   */
+  Logic.prototype.addSort = function (property, descending) {
+    var currentLevel = -1;
+    this.data.query.sort.some(function (sortObject, i) {
+      if (sortObject.property === property) {
+        currentLevel = i;
+        return;
+      }
+    });
+    if (currentLevel !== -1) { return; }
+    this.sort(property, this.data.query.sort.length, descending);
+  };
+
+  /**
+   * Remove a sort entry, shorthand of Logic.prototype.sort
+   * @param {String} property The property to remove to the sort
+   */
+  Logic.prototype.removeSort = function (property) {
+    this.sort(property, -1);
+  };
+
+
+
+  /**
+   * Get the filter descriptor associated to a property
+   * @param {String} property
+   */
+  Logic.prototype.getFilterDescriptor = function (property) {
+    if (this.data.query.properties.indexOf(property) === -1) { return; }
+    var propertyFilter = this.getPropertyDescriptor(property).filter || {};
+    var filterId = propertyFilter.id || "text";
+    // get default filter configuration, and combine it with current property
+    var filterDescriptor = this.data.meta.filters.find(function (filterDescr) {
+      return filterDescr.id === filterId;
+    });
+    $.extend(filterDescriptor, propertyFilter);
```

Comment:
It doesn't fill right to overwrite the ``filterDescriptor``, which basically contains default filter configuration. It should probably be something like:

    var filterConfig = $.extend({}, filterDescriptor, filterConfigFromPropertyType, filterConfigFromPropertyDescriptor);

Reply:
Yes, totally.